### ISSUE TITLE
False positive on ETH

### ISSUE THREAD

**Issue by dferrandizmont** [valamidev/web3-defi-honeypot-and-slippage-checker#19]
I'm trying to use the contract deployed on ETH `0xe7e07a2281f1e66e938ae7feefc69db181329f12` but when checking this token `0xda31d0d1bc934fc34f7189e38a413ca0a5e8b44f` results are: 

```
{
    "buyTax": -1,
    "sellTax": -1,
    "isHoneypot": 1,
    "error": {
        "data": null
    }
}
```

while using [honeypot.is](https://honeypot.is/ethereum?address=0xda31d0d1bc934fc34f7189e38a413ca0a5e8b44f) the result is success. What can be wrong with the contract?. Thanks for you work @valamidev !

**Comment by valamidev** [comment]
Yes, it is expected, this token in only tradeable on UniswapV3. This honeypot checker is designed for UniSwapV2, PanckakeSwap and other UniSwap forks.

To avoid to get `"isHoneypot": 1` response in this case, you should add a pre-check for your implementation that is this token listed on UniSwapV2 or V3, which is easily possible with calling the UniSwapV2 factory contract.

**Comment by dferrandizmont** [comment]
Oh cool, thanks for the answer.

But what's the real diff between router v2 and v3? Can the contract be adapted to v3?. As far as I can see both have the same methods on the interface.

**Comment by dferrandizmont** [comment]
Okey, that would be really cool! Thanks @valamidev
